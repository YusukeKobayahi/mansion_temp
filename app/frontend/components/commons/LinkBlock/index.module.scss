@import "~/styles/variables";
@import "~/styles/utils";

.item {
  color: #afafaf;
  width: 25%;
  margin-bottom: 20px;

  @include screenMd {
    width: 50%;
    margin-bottom: 0;
    border-bottom: 1px solid #d0d0d0;

    &:nth-child(odd) {
      border-right: 1px solid #d0d0d0;
    }

    &:nth-child(-n + 2) {
      border-top: 1px solid #d0d0d0;
    }
  }
}

button.link {
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  position: relative;
  color: $color-main;
  text-decoration: underline;
  background-color: unset;
  padding: 0;
  line-height: 1;

  @include screenMd {
    display: block;
    padding: 15px 18px;
    line-height: 1.3;

    &::after {
      content: "";
      width: 10px;
      height: 10px;
      border-top: 1px solid #bca565;
      border-right: 1px solid #bca565;
      transform: translateY(-50%) rotate(45deg);
      position: absolute;
      top: 50%;
      right: 18px;
    }
  }
}

.count {
  @include screenMd {
    font-size: 10px;
  }
}
