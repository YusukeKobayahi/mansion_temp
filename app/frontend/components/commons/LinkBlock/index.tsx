import Link from "next/link";
import { buildLibraryPagePath } from "~/lib/utils";
import { event } from "~/lib/gtag";
import { GoogleAnalyticsEvent } from "~/lib/types";

import styles from "./index.module.scss";

export interface LinkBlockProps {
  name: string;
  id: string;
  mansionCount: number;
  link: "cities" | "towns" | "lines" | "stations" | "schools" | "freeword";
  eventParams?: GoogleAnalyticsEvent;
  page?: string;
}

const LinkBlock: React.FC<LinkBlockProps> = ({
  name,
  id,
  mansionCount,
  link,
  eventParams,
  page,
}: LinkBlockProps) => {
  const clickLink = () => {
    if (eventParams) event(eventParams);
  };
  const clickFromPrefectures = () => {
    if (page === "prefectures") {
      sessionStorage.setItem("from", "prefectures");
    }
  };
  const href = buildLibraryPagePath(link, id);

  return (
    <>
      {mansionCount !== 0 && (
        <li key={id} className={styles.item}>
          <Link href={href} prefetch={false} passHref>
            <button
              className={styles.link}
              onClick={() => {
                clickLink();
                clickFromPrefectures();
              }}
            >
              {name}
              <span className={styles.count}>{`(${mansionCount}ä»¶)`}</span>
            </button>
          </Link>
        </li>
      )}
    </>
  );
};

export default LinkBlock;
