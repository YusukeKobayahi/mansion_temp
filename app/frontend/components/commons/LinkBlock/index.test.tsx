import { render } from "@testing-library/react";
import LinkBlock from "~/components/commons/LinkBlock";

describe("components/commons/LinkBlock", () => {
  it("正常に表示される", () => {
    const dummyData = {
      name: "千代田区",
      id: "13101",
      mansionCount: 127,
      link: "cities",
    };
    const { asFragment } = render(<LinkBlock {...dummyData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
