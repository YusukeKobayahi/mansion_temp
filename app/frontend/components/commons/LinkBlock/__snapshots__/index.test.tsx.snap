// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/LinkBlock 正常に表示される 1`] = `
<DocumentFragment>
  <li
    class="item"
  >
    <button
      class="link"
      href="/cities/13101"
    >
      千代田区
      <span
        class="count"
      >
        (127件)
      </span>
    </button>
  </li>
</DocumentFragment>
`;
